api/src/commonMain/kotlin/org/jetbrains/kotlinx/dl/api/inference/InferenceModel.kt

```diff
@@ -9,22 +9,22 @@ import org.jetbrains.kotlinx.dl.api.core.metric.Metrics
 import org.jetbrains.kotlinx.dl.dataset.Dataset
 
 /**
- * The basic class for all models which defines the basic interfaces required for inference tasks only.
+ * The basic interface for all models which defines the basic functions required for inference tasks only.
  */
-public abstract class InferenceModel : AutoCloseable {
+public interface InferenceModel : AutoCloseable {
     /** Input specification for this model. */
-    public abstract val inputDimensions: LongArray
+    public val inputDimensions: LongArray
 
     /** Model name. */
-    public var name: String? = null
+    public var name: String?
```

Comment:
Why do we need the 'name' field?

Reply:
I'm not sure. I guess, the name field was added because in keras models have a name. But we probably don't need it in the interface, so I'll remove it.